Update React Native Onboarding

smeubank opened the pull request:
# Auto (cli wizard)

There were a number of improvements which are available for other onboardings which was not present for react-native

- add options for saas, org and project slugs
- add auto and manual toggles
- moved all the manual config steps to the manual flow

<img width="1188" alt="image" src="https://github.com/user-attachments/assets/c912dd9d-4d25-4004-a5f0-5716a84a620e" />

# Manual

Manual has remained almost the same as it was except:
- removed the wizard snippet, since that doesn't make sense
- and replaced with a code box that toggles between npm, yarn, and pnpm (stolen from node :) )
- **TODO** we should update the manual setup to mirror [Expo](https://docs.sentry.io/platforms/react-native/manual-setup/expo/) and react-native (vanilla? metro?). We could add more toggles like Vue 2 and Vue 3 onboarding.

<img width="626" alt="image" src="https://github.com/user-attachments/assets/b66ac10a-4912-42aa-ac52-6b3a2efdd0e0" />

smeubank committed:
Update React Native Onboarding

There were a number of improvements which are available for other onboardings which was not present for react-native

- add options for saas, org and project slugs
- add auto and manual toggles
- moved all the manual config steps to the manual flow

krystofwoldrich commented:
The `@sentry/profiling-node` should not be in the install command.

smeubank committed:
fix test and remove profiling from isntall

smeubank commented:
> The `@sentry/profiling-node` should not be in the install command.

good spot, fixed, still toggles the profiling option in the snippet

<img width="667" alt="image" src="https://github.com/user-attachments/assets/e98755fb-ca99-480c-af5d-c58d99289a6e" />

smeubank committed:
remove unused code

smeubank committed:
Merge branch 'master' into (feat)react-native-wizard-options

krystofwoldrich commented on static/app/gettingStartedDocs/react-native/react-native.tsx:
```diff
@@ -92,253 +123,284 @@ Sentry.mobileReplayIntegration({
+            configurations: [
+              {
+                language: 'bash',
+                code: `npx @sentry/wizard@latest -s -i reactNative --org ${params.organization.slug} --project ${params.projectSlug}`,
```
The `-s` is not necessary AFAIK.

krystofwoldrich replied:
I tested the new command works as expected with RN.

Except the questions SaaS or self hosted and do you have an account? But that issue for all wizards.

smeubank replied:
```suggestion
                code: `npx @sentry/wizard@latest -i reactNative --org ${params.organization.slug} --project ${params.projectSlug}`,
```

krystofwoldrich commented on static/app/gettingStartedDocs/react-native/react-native.tsx:
```diff
@@ -92,253 +123,284 @@ Sentry.mobileReplayIntegration({
+                    <List symbol="bullet">
+                      <ListItem>
+                        {t(
+                          'Configure the SDK with your DSN and performance monitoring options'
```
The wizard doesn't do any performance changes at the moment.

````suggestion
                          'Configure the SDK with your DSN'
````

smeubank committed:
Update static/app/gettingStartedDocs/react-native/react-native.tsx

Co-authored-by: Krystof Woldrich <31292499+krystofwoldrich@users.noreply.github.com>

getsantry[bot] committed:
:hammer_and_wrench: apply pre-commit fixes

smeubank committed:
Update static/app/gettingStartedDocs/react-native/react-native.tsx

smeubank committed:
Merge branch 'master' into (feat)react-native-wizard-options

smeubank committed:
fix async causing tests to fail

smeubank committed:
Merge branch '(feat)react-native-wizard-options' of https://github.com/getsentry/sentry into (feat)react-native-wizard-options

smeubank committed:
fixes for tests

Added selectedOptions with installationMode: InstallationMode.MANUAL to ensure we see the Configure SDK section
This should fix the failing test since we're now explicitly setting the installation mode to manual, which shows all the expected sections including "Configure SDK".

smeubank committed:
dumb

krystofwoldrich reviewed (approved):
Thank you! 🚀 

Looks awesome!
